components/autocomplete/index.js

```diff
@@ -225,22 +225,44 @@ export class Autocomplete extends Component {
 		this.node = node;
 	}
 
-	insertCompletion( range, replacement ) {
-		const container = document.createElement( 'div' );
-		container.innerHTML = renderToString( replacement );
-		while ( container.firstChild ) {
-			const child = container.firstChild;
-			container.removeChild( child );
-			range.insertNode( child );
-			range.setStartAfter( child );
-		}
+	insertCompletion( range, replacement, completerName ) {
+		// Wrap completions so we can treat them as tokens with editing boundaries.
+		const tokenWrapper = document.createElement( 'span' );
```

Comment:
Don't we have a native tinymce API for this?

Reply:
There are TinyMCE APIs for this, but I'm guessing this component shouldn't depend on TinyMCE? Only the `RichText` component is aware of it. This is similar to writing flow which has tried to avoid using TinyMCE APIs. If we want we could pass the `editor` instance though.